var/spack/repos/builtin/packages/libproxy/package.py

```diff
@@ -20,17 +20,27 @@ class Libproxy(CMakePackage):
     version("0.4.13", sha256="d610bc0ef81a18ba418d759c5f4f87bf7102229a9153fb397d7d490987330ffd")
 
     variant("perl", default=False, description="Enable Perl bindings")
-    variant("python", default=True, description="Enable Python bindings", when="@0.4.16:")
+    variant("python", default=False, description="Enable Python bindings", when="@0.4.16:")
 
     depends_on("zlib")
     depends_on("perl", type=("build", "run"), when="+perl")
-    depends_on("python", type=("build", "run"), when="+python")
+
+    extends("python", when="+python")
+
+    conflicts("python@3.9:", when="@:0.4.16")
```

Comment:
```suggestion
    extends("python@:3.8", when="+python")
```
I believe this works as well